cartography/intel/aws/__init__.py

```diff
@@ -153,43 +198,32 @@ def _sync_multiple_accounts(
     exception_tracebacks = []
 
     num_accounts = len(accounts)
-
-    for profile_name, account_id in accounts.items():
-        logger.info("Syncing AWS account with ID '%s' using configured profile '%s'.", account_id, profile_name)
-        common_job_parameters["AWS_ID"] = account_id
-        if num_accounts == 1:
-            # Use the default boto3 session because boto3 gets confused if you give it a profile name with 1 account
-            boto3_session = boto3.Session()
-        else:
-            boto3_session = boto3.Session(profile_name=profile_name)
-
-        _autodiscover_accounts(neo4j_session, boto3_session, account_id, sync_tag, common_job_parameters)
-
-        try:
-            _sync_one_account(
-                neo4j_session,
-                boto3_session,
-                account_id,
-                sync_tag,
-                common_job_parameters,
-                aws_requested_syncs=aws_requested_syncs,  # Could be replaced later with per-account requested syncs
-            )
-        except Exception as e:
-            if aws_best_effort_mode:
-                timestamp = datetime.datetime.now()
-                failed_account_ids.append(account_id)
-                exception_traceback = traceback.TracebackException.from_exception(e)
-                traceback_string = ''.join(exception_traceback.format())
-                exception_tracebacks.append(f'{timestamp} - Exception for account ID: {account_id}\n{traceback_string}')
-                logger.warning(
-                    f"Caught exception syncing account {account_id}. aws-best-effort-mode is on so we are continuing "
-                    f"on to the next AWS account. All exceptions will be aggregated and re-logged at the end of the "
-                    f"sync.",
-                    exc_info=True,
-                )
-                continue
-            else:
-                raise
+    num_threads = cpu_count()
+
+    logger.info(f"AWS: Using {num_threads} threads.")
+    with ThreadPoolExecutor(max_workers=num_threads) as executor:
+        for profile_name, account_id in accounts.items():
+            with neo4j_session_factory.get_new_session() as neo4j_thread_session:
+                if num_accounts == 1:
+                    # Use the default boto3 session because boto3 gets confused if you give it a profile name w/ 1 acc.
+                    boto3_session = boto3.Session()
+                else:
+                    boto3_session = boto3.Session(profile_name=profile_name)
+
+                failure: Optional[Tuple[str, str]] = executor.submit(
+                    sync_one_account_runner,
+                    profile_name,
+                    account_id,
+                    boto3_session,
+                    neo4j_thread_session,
+                    sync_tag,
+                    common_job_parameters,
+                    aws_requested_syncs,
+                    aws_best_effort_mode,
+                ).result()
```

Comment:
I believe this will wait for each execution in turn (consecutive execution) as the call to the `result()` method of the Future returned by `executor.submit` will wait for the future to complete.
To get concurrent execution I believe you would want to submit the sync functions in a loop as is done here, collect the returned Future objects in a list and then call `concurrent.futures.wait` to wait for all of the syncs to be complete (or iterate over `concurrent.futures.as_completed` if you want to report sync statuses as they complete).